sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -154,22 +171,52 @@ def __str__(self):
         return event_str
 
     @classmethod
-    def _from_message(cls, message):
-        # type: (Message) -> EventData
+    def _from_message(cls, message, raw_amqp_message=None):
+        # type: (Message, Optional[AmqpAnnotatedMessage]) -> EventData
         """Internal use only.
 
-        Creates an EventData object from a raw uamqp message.
+        Creates an EventData object from a raw uamqp message and, if provided, AmqpAnnotatedMessage.
 
         :param ~uamqp.Message message: A received uamqp message.
+        :param ~azure.eventhub.amqp.AmqpAnnotatedMessage message: An amqp annotated message.
         :rtype: ~azure.eventhub.EventData
         """
         event_data = cls(body="")
-        event_data.message = message
+        if raw_amqp_message and message:
+            event_data._raw_amqp_message = raw_amqp_message
+            event_data.message = message
+        elif message:
+            event_data._raw_amqp_message = AmqpAnnotatedMessage(message=message)
+            event_data.message = message
         return event_data
 
     def _encode_message(self):
         # type: () -> bytes
-        return self.message.encode_message()
+        # pylint: disable=protected-access
+        return self._raw_amqp_message._message.encode_message()
+
+    def _decode_non_data_body_as_str(self, encoding="UTF-8"):
+        # type: (str) -> str
+        # pylint: disable=protected-access
+        self.raw_amqp_message._message._body._encoding = encoding
+        body = self.raw_amqp_message._message._body
+        if self.body_type == AmqpMessageBodyType.VALUE:
```

Comment:
what about the sequence type?

Reply:
sequence type returns the same for python 2.7 and 3+, so we don't need to modify the data

Reply:
updated to take seq based on discussion: seq should return string of list, not just string of all concatenated values